service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/content/HmctsEmailContentProvider.java

```diff
@@ -30,4 +44,22 @@ public HmctsEmailContentProvider(LocalAuthorityNameLookupConfiguration localAuth
             .put("localAuthority", localAuthorityNameLookupConfiguration.getLocalAuthorityName(localAuthorityCode))
             .build();
     }
+
+    public Map<String, Object> buildC2UploadNotification(final CaseDetails caseDetails) {
+        CaseData caseData = objectMapper.convertValue(caseDetails.getData(), CaseData.class);
+
+        List<Map<String, Object>> respondents1 = (ObjectUtils.isEmpty(caseDetails.getData().get("respondents1"))
```

Comment:
Not sure why caseData was created above if it's not used (or rather used to get the family man?)

Couldn't the code to get respondents to be refactored to use caseData? I find it confusing and difficult to read, at least extract it to another method please.

Reply:
updated